File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        // Arrays.sort(nums);
        int n = nums.length;
        Set<Integer> hashset = new HashSet<>();

        for(int i = 0;i<n;i++){
            if(hashset.contains(nums[i])){
                return true;
            }else{
                hashset.add(nums[i]);
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n = nums.size();
        vector<bool>check(n+1);
        for(int x:nums){
            if(check[x-1]){
                return x;
            }else{
                check[x-1] = true;
            }
        }

        return 0;
    }
};
